### ISSUE TITLE
test suite failures with development node due to missing experimental-wasm-bigint

### ISSUE THREAD

**Issue by wingo** [emscripten-core/emscripten#13576]
Steps to repeat:

1. build fresh llvm, binaryen, node from git.
2. run test suite, e.g.:

```
LLVM=/home/wingo/src/llvm-project/build/bin BINARYEN=/home/wingo/src/binaryen/ NODE=/home/wingo/src/node/out/Release/node tests/runner 'wasm*'
```

expected results: everything passes

actual results:
```
======================================================================
FAIL: test_utf8_textdecoder (test_core.wasmz)
----------------------------------------------------------------------
[...]
AssertionError: JS subprocess failed (/home/wingo/src/node/out/Release/node --experimental-wasm-bigint benchmark_utf8.js): 9.  Output:
/home/wingo/src/node/out/Release/node: bad option: --experimental-wasm-bigint
```

Indeed latest node git doesn't support `--experimental-wasm-bigint`.  Downgrading to `v14.16.0` does the trick though.

**Comment by kripken** [comment]
Interesting. It looks like node/v8 simply removed the flag, I guess after it was considered stable?

Sadly we can't just switch to using `--wasm-staging` as that flag does not enable bigint on older node yet. So I'm not sure what's the best thing here. Version-checking node in the test runner seems error-prone, but not sure what else we can do...

**Comment by wingo** [comment]
Perhaps we should do some node version-checking regardless.  Here when I was getting test suites to run, I first tried with the node from this Ubuntu 20.04 LTS system (v10.19.0), but it's too old as it lacks TextDecoder, so there were quite a number of test failures.

**Comment by sbc100** [comment]
Agreed.   We can be pretty strict about the precise version we use in tests.   No need to support a wide range here.   Our output needs to support a wide range but we only need to support running the tests against one version at a time.   

This is even more true for our internal node tools (e..g compiler.js which can and should assume they are running on the exact version of node that we ship in emsdk IMHO).

**Comment by sbc100** [comment]
@wingo .. using emsdk to install most of the emscirpten components is an easy way to avoid incompatibility and save time.   

The way I work is to install everything with emsdk then get emscripten from git if I want to work on that.   Or if I'm working on LLVM i set `EM_LLVM_ROOT` in the environment to override that one config setting.

**Comment by truboxl** [comment]
Unfortunately this impact downstream for platforms where emsdk does not cover. Even using the latest nodejs-lts (v16.16.0 as of writing) is too new and fails the tests...

Some other errors:
```
test_stdio_locking (test_core.core0) ... FAIL
-- begin program out --
/data/data/com.termux/files/usr/bin/node: bad option: --experimental-wasm-bulk-memory
-- end program output --
```

**Comment by sbc100** [comment]
I believe this should now be fixed with #18070
